model/common_structures.smithy

```diff
@@ -59,3 +59,34 @@ structure Hits{
     _source: Document,
     fields: Document
 }
+
+
+structure PitSegments {
+    index: Index,
+
+    shard: ShardNumber,
+
+    prirep: Pri,
+
+    ip: IpAddress,
+
+    segment: SegmentId,
+
+    generation: GenerationNumber,
+
+    docs_count: DocsCount,
+
+    docs_deleted: DocsDeleted,
+
+    size: SizeSegment,
+
+    size_memory: Size,
+
+    committed: Commit,
+
+    searchable: Search,
```

Comment:
For booleans like `committed`, `searchable` and `compound`, you can directly mark them here as boolean rather than defining them in the common_booleans file as they are used only for this API.

Same goes for other parameters as well which are not common.